### PR TITLE
Client: Address comments in #191

### PR DESCRIPTION
**Pull request by cfcs**
cc @olleolleolle @linse  @hannesm

### CODE REVIEW

File: test/client.ml [hunk 1]
```diff
@@ -68,19 +67,19 @@ module Parse_response_tests = struct
...
       "fails to unpack response with mismatching query ID", `Quick, fails_to_unpack_mismatched;
   ]
 end
 
-(* {!Transport} provides the implementation of the underlying flow
-   that is in turn used by {!Dns_client.Make} to provide the
-   blocking Unix convenience module:
+(* {!Transport} provides a mock implementation of the transport used by
+   Dns_client.Make. The mock data is passed as type flow and io_addr in
+   connect/recv/send by suppling the optional ?nameserver argument.
```
Comment (olleolleolle) [inline test/client.ml]: Supplying

Comment (hannesm) [inline test/client.ml]:
```suggestion
   connect/recv/send by supplying the optional ?nameserver argument.
```

File: client/dune [hunk 2]
```diff
@@ -4,3 +4,10 @@
   (modules     dns_client)
   (libraries   dns.cache domain-name dns randomconv rresult)
   (wrapped     false))
+
```
Comment (hannesm) [inline client/dune]:
```suggestion

```

File: client/dune [hunk 3]
```diff
@@ -4,3 +4,10 @@
...
+(library
```
Comment (hannesm) [inline client/dune]:
```suggestion

```

File: client/dune [hunk 4]
```diff
@@ -4,3 +4,10 @@
...
+  (name        dns_client_tls)
```
Comment (hannesm) [inline client/dune]:
```suggestion

```

File: client/dune [hunk 5]
```diff
@@ -4,3 +4,10 @@
...
+  (public_name dns-client.tls)
```
Comment (hannesm) [inline client/dune]:
```suggestion

```

File: client/dune [hunk 6]
```diff
@@ -4,3 +4,10 @@
...
+  (modules     dns_client_tls)
```
Comment (hannesm) [inline client/dune]:
```suggestion

```

File: client/dune [hunk 7]
```diff
@@ -4,3 +4,10 @@
...
+  (libraries   domain-name mirage-flow dns dns-client randomconv rresult tls)
```
Comment (hannesm) [inline client/dune]:
```suggestion

```

File: client/dune [hunk 8]
```diff
@@ -4,3 +4,10 @@
...
+  (wrapped     false))
```
Comment (hannesm) [inline client/dune]:
```suggestion

```

File: client/dune (hunk 3 above)
Comment (cfcs) [reply client/dune]: What is the suggestion here, I don't think I understand?

File: client/dune (hunk 8 above)
Comment (cfcs) [reply client/dune]: maybe this whole thing should just be commented out

**Comment by cfcs** [comment]
Thanks, I think I addressed the suggestions :) @olleolleolle @hannesm

File: client/dune (hunk 3 above)
Comment (hannesm) [reply client/dune]: to remove the dns_client_tls library which causes build failures, sorry for the obfuscated github comments.  and thanks for pushing such a commit which removes that anyways! :)

**Pull request merged**